app-modules/ai/src/Filament/Pages/Assistant/Concerns/CanManageThreads.php

```diff
@@ -333,12 +335,21 @@ public function cloneThreadAction(): Action
                     ->options(function (Get $get): Collection {
                         return match ($get('targetType')) {
                             AiThreadShareTarget::Team->value => Team::orderBy('name')->pluck('name', 'id'),
-                            AiThreadShareTarget::User->value => User::whereKeyNot(auth()->id())->orderBy('name')->pluck('name', 'id'),
+                            AiThreadShareTarget::User->value => User::tap(new WithoutSuperAdmin())->whereKeyNot(auth()->id())->orderBy('name')->pluck('name', 'id'),
                         };
                     })
                     ->searchable()
                     ->multiple()
-                    ->required(),
+                    ->required()
+                    ->rules([
+                        fn (): Closure => function (string $attribute, $value, Closure $fail) {
+                            foreach ($value as $v) {
+                                if (User::findOrFail($v)->hasRole('authorization.super_admin')) {
+                                    $fail('Super admin users cannot have a thread shared with them.');
+                                }
+                            }
+                        },
```

Comment:
Actually, I know we did it this way in the other PRs. But could you look into changing this into a custom Rule class?

I believe the same concept as in Filament Forms applies here to Actions as well: https://filamentphp.com/docs/3.x/forms/validation#custom-rules

This way the logic is a little more confined and so in the tests I think we may be able to check that the specific rule was triggered in `assertHasActionErrors` rather than rely on  `assertSeeText`, which does not actually know that the text it reads is from the validation error or not.

Reply:
Hi,
I have made one common validation rule file 'RestrictSuperAdmin' and used in both email and clone with type parameter. 

Also, Added assertHasActionErrors with error message in test case.